marshmallow_autoschema/schema_factory.py

```diff
@@ -70,79 +70,190 @@ def kwsift(kw, f):
     return out
 
 
-def get_schema_cls_name(model_cls):
+def get_schema_cls_name(model_cls: type) -> str:
     return model_cls.__name__ + 'Schema'
 
 
-def is_model_init(init):
+def get_schema_cls(model_cls: type) -> type:
+    '''
+    Returns the autogenerated schema class from an autoschemad model class.
+
+    Args:
+        model_cls:
+            The class object of the model to find the schema for.
+
+    Returns:
+        The schema class object associated with the model class.
+
+    '''
+    sn = get_schema_cls_name(model_cls)
+    try:
+        return getattr(model_cls, SCHEMA_ATTRNAME)
+    except AttributeError:
+        raise ValueError(
+            '''{} does not appear to be a valid model, as it '
+            does not have an autogenerated Schema. Expected to find '
+            {} attribute, did not.'''.format(model_cls, sn))
+
+
+def is_model_init(init: Callable[..., None]) -> bool:
     '''
     Check if an __init__ callable correponds to one monkeypatched by
     a schema factory.
     '''
     return '_mro_offset' in signature(init).parameters.keys()
 
 
-def is_type(typ, *typs):
-    return any([typ.__name__ == t.__name__ for t in typs])
+def check_type(typ: type, *typs: type) -> bool:
+    '''
+    Checks by name whether a given type is one of several options
 
+    Args:
+        typ: the type to check
+        *typs: the types to check against
+    Returns:
+        True if `typ` matches any type in `typs`.
+    '''
+    try:
+        return any([typ.__name__ == t.__name__ for t in typs])
```

Comment:
any(typ.__name__ == t.__name__ for t in typs) alone works and avoid creating an intermediate list.